src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -527,7 +543,7 @@ internal void WriteCentralDirectoryFileHeader()
             writer.Write(_fileComment != null ? (ushort)_fileComment.Length : (ushort)0); // file comment length
             writer.Write((ushort)0); // disk number start
             writer.Write((ushort)0); // internal file attributes
-            writer.Write((uint)0); // external file attributes
+            writer.Write(_externalFileAttr); // external file attributes
```

Comment:
Is this line actually getting hit by your tests? Seems you're just reading zips you already made, and testing the set/get pairs.